src/bp-members/classes/class-bp-members-notice.php

```diff
@@ -0,0 +1,758 @@
+		}
+
+		// Activate the notice.
+		$this->priority = $previous_priority;
+		$activated      = $this->save();
+
+		if ( is_wp_error( $activated ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deactivates a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @return bool
+	 */
+	public function deactivate() {
+		// Used to restore the priority on re-activation.
+		$previous_priority = $this->priority;
+		bp_notices_update_meta( $this->id, 'previous_priority', $previous_priority );
+
+		// Deactivating is using a priority of 127.
+		$this->priority = 127;
+		$deactivated    = $this->save();
+
+		if ( is_wp_error( $deactivated ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deletes a notice.
+	 *
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @return bool
+	 */
+	public function delete() {
+		global $wpdb;
+
+		/**
+		 * Please do not use this filter anymore.
+		 *
+		 * @since 1.0.0
+		 * @deprecated 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the message notice item being deleted.
+		 */
+		do_action_deprecated( 'messages_notice_before_delete', array( $this ), '15.0.0', 'bp_members_notice_before_delete' );
+
+		/**
+		 * Fires before the notice item has been deleted.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the notice item being deleted.
+		 */
+		do_action( 'bp_members_notice_before_delete', $this );
+
+		$bp  = buddypress();
+		$sql = $wpdb->prepare( "DELETE FROM {$bp->members->table_name_notices} WHERE id = %d", $this->id );
+
+		if ( ! $wpdb->query( $sql ) ) {
+			return false;
+
+			// Remove all corresponding notice metadata.
+		} else {
+			bp_notices_delete_meta( $this->id );
+		}
+
+		/**
+		 * Please do not use this filter anymore.
+		 *
+		 * @since 2.8.0
+		 * @deprecated 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the notice being saved. Passed by reference.
+		 */
+		do_action_deprecated( 'messages_notice_after_delete', array( $this ), '15.0.0', 'bp_members_notice_after_delete' );
+
+		/**
+		 * Fires after the notice item has been deleted.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the notice item being deleted.
+		 */
+		do_action( 'bp_members_notice_after_delete', $this );
+
+		return true;
+	}
+
+	/** Static Methods ********************************************************/
+
+	/**
+	 * Pulls up a list of notices.
+	 *
+	 * To get all notices, pass a value of -1 to pag_num.
+	 *
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param array $args {
+	 *     An array of arguments.
+	 *
+	 *     @type integer      $user_id          The user ID to get notices for. Defaults to `0`.
+	 *     @type integer      $pag_num          Number of notices per page. Defaults to `20`.
+	 *     @type integer      $pag_page         The page number to retrieve.  Defaults to `1`.
```

Comment:
Can get change this to a more familiar syntax? Like `$per_page` and `$page`.

Reply:
I agree we should probably move to this syntax, but then we should do it for all BuddyPress objects. I think it's best to deal with it globally from another ticket.